spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkExpressions.java

```diff
@@ -0,0 +1,322 @@
+    }
+    return lit.value();
+  }
+
+  private static Pair<Transform, String> convertAttr(Expression expr) {
+    Transform type = TRANSFORMS.get(expr.getClass());
+    if (type == Transform.IDENTITY) {
+      Attribute attr = (Attribute) expr;
+      return Pair.of(type, attr.name());
+    } else if (expr instanceof Cast) {
+      Cast cast = (Cast) expr;
+      if (DateType$.MODULE$.sameType(cast.dataType()) && cast.child() instanceof Attribute) {
+        Attribute attr = (Attribute) cast.child();
+        return Pair.of(Transform.DAY, attr.name());
+      }
+    } else if (expr instanceof ParseToDate) {
+      ParseToDate toDate = (ParseToDate) expr;
+      if (toDate.left() instanceof Attribute) {
+        Attribute attr = (Attribute) toDate.left();
+        return Pair.of(Transform.DAY, attr.name());
+      }
+    } else if (expr instanceof UnaryExpression) {
+      UnaryExpression func = (UnaryExpression) expr;
+      if (func.child() instanceof Attribute) {
+        Attribute attr = (Attribute) func.child();
+        return Pair.of(type, attr.name());
+      }
+    }
+    return null;
+  }
+
+  private static long yearToTimestampMicros(int year) {
+    return ChronoUnit.MICROS.between(EPOCH,
+            LocalDateTime.of(year, 1, 1, 0, 0).atOffset(ZoneOffset.UTC));
+  }
+
+  private static long dayToTimestampMicros(int daysFromEpoch) {
+    return ChronoUnit.MICROS.between(EPOCH,
+            EPOCH_DAY.plusDays(daysFromEpoch).atStartOfDay().atOffset(ZoneOffset.UTC));
+  }
+
+  private static org.apache.iceberg.expressions.Literal<Long> tsLiteral(long timestampMicros) {
+    return org.apache.iceberg.expressions.Literal
+            .of(timestampMicros)
+            .to(TimestampType.withoutZone());
+  }
+
+  private static org.apache.iceberg.expressions.Expression filter(
+      Operation op, String name, int value, Function<Integer, Long> startTsMicros) {
+    switch (op) {
+      case LT:
+        return predicate(Operation.LT, name, tsLiteral(startTsMicros.apply(value)));
+      case LT_EQ:
+        return predicate(Operation.LT, name, tsLiteral(startTsMicros.apply(value + 1)));
+      case GT:
+        return predicate(Operation.GT_EQ, name, tsLiteral(startTsMicros.apply(value + 1)));
+      case GT_EQ:
+        return predicate(Operation.GT_EQ, name, tsLiteral(startTsMicros.apply(value)));
+      case EQ:
+        return and(
+                predicate(Operation.GT_EQ, name, tsLiteral(startTsMicros.apply(value))),
+                predicate(Operation.LT, name, tsLiteral(startTsMicros.apply(value + 1)))
+        );
+      case NOT_EQ:
+        return or(
+                predicate(Operation.GT_EQ, name, tsLiteral(startTsMicros.apply(value + 1))),
+                predicate(Operation.LT, name, tsLiteral(startTsMicros.apply(value)))
+        );
+      case IN:
+      case NOT_IN:
+      default:
+        throw new IllegalArgumentException("Cannot convert operation to year filter: " + op);
+    }
+  }
+
+  private enum Transform {
```

Comment:
I have referred this code from 
https://github.com/Netflix/iceberg/blob/master/spark/src/main/java/com/netflix/iceberg/spark/SparkExpressions.java

Currently testcases are passing as the filter conversion is success.

I know that we can add some more transform here.
But for the case of rewrite data file procedure , the filters will be always `identity`  transform as it comes from spark ? So, no need of other transform ?